tutorials/cloudsdk-integrate-cap/cloudsdk-integrate-cap.md

```diff
@@ -0,0 +1,470 @@
+
+        service.createBusinessPartner(bp).execute(httpDestination);
+    }
+
+    private List<CapBusinessPartner> convertS4BusinessPartnersToCapBusinessPartners(
+            final List<BusinessPartner> s4BusinessPartners,
+            final String destinationName) {
+        final List<CapBusinessPartner> capBusinessPartners = new ArrayList<>();
+
+        for (final BusinessPartner s4BusinessPartner : s4BusinessPartners) {
+            final CapBusinessPartner capBusinessPartner = com.sap.cds.Struct.create(CapBusinessPartner.class);
+
+            capBusinessPartner.setFirstName(s4BusinessPartner.getFirstName());
+            capBusinessPartner.setSurname(s4BusinessPartner.getLastName());
+            capBusinessPartner.setId(s4BusinessPartner.getBusinessPartner());
+            capBusinessPartner.setSourceDestination(destinationName);
+
+            capBusinessPartners.add(capBusinessPartner);
+        }
+
+        return capBusinessPartners;
+    }
+}
+```
+
+[CDS Query Notation](https://cap.cloud.sap/docs/cds/cqn) (CQN) is the common language in CAP to run queries against services. It can be used to talk to the services defined by your model, but also remote services, such as the database.
+
+The event handler uses the following APIs, which are available for service providers in CAP Java:
+
+* [Event handler classes](https://cap.cloud.sap/docs/java/srv-impl#event-handler-classes) have to implement the marker interface `EventHandler` and register themselves as Spring Beans (`@Component`). The marker interface is important, because it enables the CAP Java runtime to identify these classes among all Spring Beans.
+* [Event handler methods](https://cap.cloud.sap/docs/java/srv-impl#event-handler-methods) are registered with `@On`, `@Before`, or `@After` annotations. Every event, such as an entity creation, runs through these three [phases](https://cap.cloud.sap/docs/java/srv-impl#event-phases). Each phase has a slightly different semantic.
+* The annotation `@ServiceName` specifies the default service name all event handler methods apply to. Here this is `cloud.sdk.capng`, as this was also the name when defining the service in the CDS model.
+
+[DONE]
+[ACCORDION-END]
+
+[ACCORDION-BEGIN [Step 6: ](Understanding the read and write methods)]
+
+The above class handles the READ and CREATE events (highlighted above).
+
+- The READ operation just returns the business partners from the remote OData service.
+
+    Here we are using the OData VDM from SAP Cloud SDK to fetch business partners from the target system. We can call this service `getAllBusinessPartner()` function to retrieve a list of all the business partners from the system. In this tutorial our destination is called `MyErpSystem` and you will learn a little later where it is set up.
+
+    Later we transform `BusinessPartner` to `CapBusinessPartner` according to our requirements.
```

Comment:
Always say "you will" or "your system", never "we".

Please change all occurrences accordingly.